lib/pages/main/wallet_contents/add_wallet_connect/add_wallet_connect.dart

```diff
@@ -0,0 +1,333 @@
+import 'dart:async';
+import 'dart:developer';
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:mobile_scanner/mobile_scanner.dart';
+import 'package:qubic_wallet/components/wallet_connect/pair.dart';
+import 'package:qubic_wallet/config.dart';
+import 'package:qubic_wallet/di.dart';
+import 'package:qubic_wallet/flutter_flow/theme_paddings.dart';
+import 'package:qubic_wallet/helpers/global_snack_bar.dart';
+import 'package:qubic_wallet/helpers/platform_helpers.dart';
+import 'package:qubic_wallet/l10n/l10n.dart';
+import 'package:qubic_wallet/pages/main/wallet_contents/add_wallet_connect/components/scanner_corners_border.dart';
+import 'package:qubic_wallet/pages/main/wallet_contents/add_wallet_connect/components/scanner_overlay_clipper.dart';
+import 'package:qubic_wallet/services/wallet_connect_service.dart';
+import 'package:qubic_wallet/stores/application_store.dart';
+import 'package:qubic_wallet/styles/button_styles.dart';
+import 'package:qubic_wallet/styles/edge_insets.dart';
+import 'package:qubic_wallet/styles/input_decorations.dart';
+import 'package:qubic_wallet/styles/text_styles.dart';
+import 'package:qubic_wallet/styles/themed_controls.dart';
+import 'package:reown_walletkit/reown_walletkit.dart';
+
+part 'components/add_wallet_connect_desktop_view.dart';
+part 'components/add_wallet_connect_mobile_view.dart';
+
+enum DomainType { valid, unknown, scam, mismatch }
+
+class AddWalletConnect extends StatefulWidget {
+  const AddWalletConnect({super.key});
+
+  @override
+  // ignore: library_private_types_in_public_api
+  _AddWalletConnectState createState() => _AddWalletConnectState();
+}
+
+class _AddWalletConnectState extends State<AddWalletConnect> {
+  final ApplicationStore appStore = getIt<ApplicationStore>();
+  final GlobalSnackBar _globalSnackBar = getIt<GlobalSnackBar>();
+  final WalletConnectService walletConnectService =
+      getIt<WalletConnectService>();
+  final TextEditingController urlController = TextEditingController();
+
+  bool isLoading = false;
+  Timer? pairingTimer;
+  StreamSubscription<SessionProposalEvent?>? sessionProposalSubscription;
+  StreamSubscription<SessionProposalErrorEvent?>?
+      sessionProposalErrorSubscription;
+
+  int? wcPairingId;
+  PairingMetadata? wcPairingMetadata;
+  List<String> wcPairingMethods = [];
+  List<String> wcPairingEvents = [];
+  Map<String, Namespace>? wcPairingNamespaces;
+  @override
+  void initState() {
+    super.initState();
+    walletConnectService.initialize().then((value) {
+      final l10n = l10nOf(context);
+
```

Comment:
@Qubic-Hub as all errors are handled in the same way, please do not duplicate the code for each if

Reply:
refactored code